### PR TITLE
Feature/keeper script

### PR DESCRIPTION
**Pull request by Chepelau**
Simple script designed to be run with cron/cloud funcs

### CODE REVIEW

File: scripts/keeper.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    parser.add_argument(
+        '--function_arguments', '-fa', action='append', default = []
+    )
+    return parser
+
+def get_abi(abi_path: str) -> List:
+    with open(abi_path, 'r') as f:
+        abi = json.load(f)
+
+    return abi
```
Comment (DaveVodrazka) [inline scripts/keeper.py:64]: Maybe sprinkle in a few try catches? 😄 
```suggestion
def get_abi(abi_path: str) -> List:
    try:
        f = open(abi_path, 'r')
    except:
        raise ValueError("Invalid ABI path")
    with f:
        try:
            abi = json.load(f)
        except:
            raise ValueError("Invalid ABI file")

    return abi
```

**Comment by tensojka** [comment]
For now, high-level feedback: IME the biggest danger/pain – the only thing that's really an issue if it doesn't work – is status/error reporting. What worked well for me is Telegram Bot API, it's super simple and never failed on me. I'd do this _in addition_ to whatever GCF are offering for monitoring.

```
def alert(msg: str, alert=False):
    # https://api.telegram.org/bot[BOT_API_KEY]/sendMessage?chat_id=[MY_CHANNEL_NAME]&text=[MY_MESSAGE_TEXT]
    if alert:
        chat_id = alert_chat_id
    else:
        chat_id = status_chat_id
    params = {
        'chat_id': chat_id,
        'text': msg,
    }
    logger.info(msg)
    res = requests.get("https://api.telegram.org/bot" + BOT_API_KEY + "/sendMessage", params=params)
    logger.debug(res.text)
    res.raise_for_status()
```

I'd go as far as this

```
except Exception as e:
        # https://stackoverflow.com/a/54083435/2997134
        full_traceback = "".join(traceback.format_exception(e, value=e, tb=e.__traceback__))
        alert("Exiting, encountered fatal error: " + str(e) + ". \n" + full_traceback, True)
```

**Comment by Chepelau** [comment]
Isn't there some functionality on g funcs to send a notification if stderr returns something? Like cron with MAILTO(i think)?

**Comment by tensojka** [comment]
Don't know 🤷

**Comment by DaveVodrazka** [comment]
> Also, isn't there some functionality on g funcs to send a notification if stderr returns something? Like cron with MAILTO(i think)?

GCP suggests using this https://github.com/GoogleCloudPlatform/python-docs-samples/blob/HEAD/functions/helloworld/main.py#L190-L204

But that only stores the error in their error reports, so maybe combination of this and telegram BOT?

**Comment by tensojka** [comment]
@DaveVodrazka Looks like the setup will be:

Setup bude Cloud Scheduler -> Cloud Functions -> Cloud Logging and from there: https://cloud.google.com/monitoring/support/notification-options#mobile-app

(Alongside Telegram)

**Pull request closed without merging**